### ISSUE TITLE
Mark everything as watched

### ISSUE THREAD

**Issue by ChaotikZ** [damienhaynes/TraktRater#57]
I've been using IMDb for years now, probably since around the time I first got Internet.
Usually, when I finish watching a movie, I mark them but don't rate them, so they are all part of my watchlist.

My problem is, if I try to export them using TraktRater, all I get is a "Watchlist" list on Trakt, and the only way to mark them as watched is either by rating all those entries on IMDb (3000+ movies) and exporting it Trakt or importing them onto Trakt and mark them as watched there.

I was wondering if there was a simpler way to do this, or if it's even possible to add the option to mark everything on a wishlist/list as watched using TraktRater.

**Comment by ssmb212** [comment]
Greetings what Chaotikz is trying to say the imdb watchlist is being imported as a personal list to trakt.tv specifically as a "to watch watchlist" is there any way to import it to as trakt.tv "watched history" or just mass transfer personal watchlist to watched history on trakt.tv?

**Comment by damienhaynes** [comment]
It doesn't make sense to set IMDb Watchlist -> Trakt Watched History. Since there is a 1:1 mapping of Watchlist i.e. what you intend to watch some time in the future.

If you have a rated csv, then it implies you have watched it from the fact you have rated it and therefore we have an option to not only import your ratings but also mark it as watched.

I don't think it makes sense to add a setting to mark as watched for the watchlist, but I could do something for a custom list...I will have to think about it.

**Comment by ssmb212** [comment]
the thing is that many of us we use trakt.tv on android apps or our smart tv's and when we want to watch something it shows as watched on these apps only if its on "watched history"list, also having 4300 watched movies on imdb while i only have rated 160 of them.

**Comment by damienhaynes** [comment]
So do you have a custom list which has your 4300 movies watched on IMDb? or are they part of your Watchlist on IMDb?

**Comment by ssmb212** [comment]
they are part of imdb watchlist (as watched) i exported list as .csv imported to trakt.tv and they all went fine but they are under personal list/watchlist.

**Comment by damienhaynes** [comment]
Ok, thats fine. Its working as intended because thats what the watchlist is. 

I will add support for custom csv import in a future version where you can use any csv file to import what you want basis the contents of the file or apply a global setting such as 'watched' in your case.

**Comment by ssmb212** [comment]
well the default watchlist of imdb is the already watched ones, i took some screenshots with the issue. I wouldnt bother but the only thing that works as "watched" on apps is the watched history from trakt.tv

Here is a screenshot with what i mean in detail.
Pretty much id like to transfer my watchlist+watched status to trakt.tv so i know in the apps that work with trakt.tv what i have watched and what not makes sense?

![screenshot 91](https://user-images.githubusercontent.com/33333729/32357272-9bcbfc48-c03b-11e7-9275-c0c1e33dba66.png)

**Comment by damienhaynes** [comment]
I think thats a difference between watchlist behaviour on trakt.tv. Where if something is on your watchlist then you watch it, it automatically removes it.

Okay, so can you attach your watchlist csv from IMDb here. I want to see if it reports if the item is watched. If it does, then I can use that to also mark something as watched. 

I could also look at supporting marking as watched for watchlist items on IMDb. The question would be should it also add to trakt.tv watchlist as well in that case or just treat them all as watched if the user wants that. Let me see what your CSV looks like first.

**Comment by ssmb212** [comment]
Here you go both watchlist and ratings, thanks a lot for the help.
[imdb.zip](https://github.com/damienhaynes/TraktRater/files/1439830/imdb.zip)

**Comment by damienhaynes** [comment]
Typical, they dont include the fact that you have watched it in the Watchlist CSV. So there goes that idea out the window. Only way would be to have an option to:

1. Mark all Watchlisted items as watchlisted only (Default, how it works now);
2. Mark all Watchlisted items as watched and watchlisted;
3. Mark all Watchlisted items as watched only.

Not really a good outcome since you may not have watched all the items on your watchlist which I would assume would be the most common scenario. But it would be your choice to do that.

**Comment by ssmb212** [comment]
i have watched all my items, can you tell me how to do that ?

**Comment by damienhaynes** [comment]
There is no way to import it like that currently because I have not provided an option yet.

IMDb should really include that information in their CSV export, but they dont :(

**Comment by ssmb212** [comment]
Is there a way to like transfer the items from watchlist to rating list? and like mark them all as a 10/10 rating or w/e maybe then if imported the rating list with the 4300 movies would go into the history watched list?

**Comment by damienhaynes** [comment]
Yes, what you can do is open your Watchlist CSV. And add 10 to the 'You rated' column for each 4300 movies you have.

When that is done, you can import it as a rating CSV and it will mark the items as rated 10/10 and watched. It will only mark items as watched if rated if you have enabled that option in the Options tab.

Also another thing which you should do is ensure you switch to the oAuth authentication and not username/password if you are still using that. trakt.tv is no longer allowing updating the users watch history when using old username/password authentication.

**Comment by ssmb212** [comment]
oh i see i was using password and username.

**Comment by ssmb212** [comment]
ill give that a try and post back with the results.

**Comment by ssmb212** [comment]
It indeed ended up working, it took a while to finish and also i had to only import rating list (witch was my watchlist) alone if i used both fields rating and watching lists i was getting errors.

**Comment by dafzor** [comment]
Why not just add an option to mark everything that's imported as watched without having to create fake ratings? Seems like it would be fairly straight forward to do.

**Comment by kajh** [comment]
When I import a serie where I have rated the serie but not the individual episodes I dont get added to my watched history at trakt.tv. Any ideas how make the whole serie added to my watched history?

**Comment by damienhaynes** [comment]
Only rated episodes can be marked as watched. A lot of people would only rate a show after one season watched but there could be many seasons for a show.
